modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/RankEvalSpec.java

```diff
@@ -120,9 +152,51 @@ public void setSpecifications(Collection<RatedRequest> specifications) {
         } , REQUESTS_FIELD);
     }
 
+    public static RankEvalSpec parse(XContentParser parser, RankEvalContext context, boolean templated) throws IOException {
+        RankEvalSpec spec = PARSER.parse(parser, context);
+
+        if (templated) {
+            logger.trace("template: {}",  spec.script);
+            for (RatedRequest query_spec : spec.getSpecifications()) {
+                @SuppressWarnings({ "unchecked", "rawtypes" })
+                Map<String, String> params = (Map) query_spec.getParams();
+                logger.trace("params: {}", params);
+                Script scriptWithParams = new Script(spec.script.getScript(), spec.script.getType(), spec.script.getLang(), params);
+                String resolvedRequest = 
+                        ((BytesReference) 
+                                (context.getScriptService().executable(scriptWithParams, ScriptContext.Standard.SEARCH, params)
+                                        .run()))
+                        .utf8ToString();
+                try (XContentParser subParser = XContentFactory.xContent(resolvedRequest).createParser(resolvedRequest)) {
+                    QueryParseContext parseContext =
+                            new QueryParseContext(
+                                    context.getSearchRequestParsers().queryParsers, 
+                                    subParser, 
+                                    context.getParseFieldMatcher());
+                    if (resolvedRequest != null) {
```

Comment:
I can't think of any valid condition where this could be null actually. Removing.